ProcessLib/VectorMatrixAssembler.cpp

```diff
@@ -114,17 +136,32 @@ void VectorMatrixAssembler::assembleWithJacobian(
     }
     else
     {
-        auto local_coupled_xs0 = getPreviousLocalSolutions(*cpl_xs, indices);
-        auto local_coupled_xs = getCurrentLocalSolutions(*cpl_xs, indices);
-
-        ProcessLib::LocalCoupledSolutions local_coupled_solutions(
-            cpl_xs->dt, cpl_xs->process_id, std::move(local_coupled_xs0),
-            std::move(local_coupled_xs));
-
-        _jacobian_assembler->assembleWithJacobianAndCoupling(
-            local_assembler, t, local_xdot, dxdot_dx, dx_dx, _local_M_data,
-            _local_K_data, _local_b_data, _local_Jac_data,
-            local_coupled_solutions);
+        if (base_dof_table == nullptr)
```

Comment:
Needs discussion: Does it also cover the T weakly coupled to HM case?

Reply:
It is gone. Now a vector of DOF tables is passed to this function, which makes coupling more flexible. See the commit entitled with
``
[Coupling] Passed a vector of DOF tables to VectorMatrixAssembler ``